packages/metadata-editor/src/MetadataEditor.tsx

```diff
@@ -192,74 +183,124 @@ export class MetadataEditor extends ReactWidget {
     }
   }
 
-  onAfterShow(): void {
-    if (!this.editor) {
+  onUpdateRequest(msg: Message): void {
+    super.onUpdateRequest(msg);
+    if (!this.editor && this.metadata['code']) {
```

Comment:
to confirm, this implementation still only supports a single code field per metadata for now. I'm ok with that for an initial commit to get this in as long as we create a follow up to address it later

Reply:
Yes. This should definitely be addressed in a future PR but since this is high priority and we only have code snippets so far I think it can wait.